src/coreclr/md/runtime/strongnameinternal.cpp

```diff
@@ -6,20 +6,28 @@
 
 #include "stdafx.h"
 #include "strongnameinternal.h"
+#include "sha1.h"
+
+// We allow a special abbreviated form of the Microsoft public key (16 bytes
+// long: 0 for both alg ids, 4 for key length and 4 bytes of 0 for the key
+// itself). This allows us to build references to system libraries that are
+// platform neutral (so a 3rd party can build SPCL replacements). The
+// special zero PK is just shorthand for the local runtime's real system PK,
+// which is always used to perform the signature verification, so no security
+// hole is opened by this. Therefore we need to store a copy of the real PK (for
+// this platform) here.
 #include "thekey.h"
```

Comment:
You can simplify it even more and delete these header files. These keys are only ever used in the PublicKey -> PublickKeyToken conversions as a cache that skips over the more expensive SHA1 computation. These PublicKey and PublicKeyToken can be an internal implementation details of the PublicKey -> PublickKeyToken conversion method. They do not need to be distributed over multiple header files.

(While you are on it, you can also check that we have the cache pairs for all PublicKeys used by our shared frameworks. I won't be surprised if we have missed some. This cache should be good for startup perf.)